### PR TITLE
Quest clean up & display fixes

### LINKED ISSUE #635: [Team 5] Fixing quest implementation and accounting for biomes
# **Description**
**Task:** Fixing quest implementation and accounting for biome .
**Feature:** [Quests](#67)

Updating the quest design to better suit existing biome setup and fix non working quests.
 
# **Dependencies**
- [x] Quest Completion and System Enhancements (#516)

# **Milestones**
List of steps that need to be completed for this task.
- [x] Changing quest design (NPC, enemies and items) to suit biome setup (Oct. 16)
- [x] Adding mini game quests progression triggers (Oct. 16)
- [x] Fixing quest order to suit final boss summoning triggers (Oct. 16)
- [x] Adding final boss summoning triggers (overlooked last sprint) (Oct. 16)
- [x] Fixing final boss defeat quest triggers (not working currently) (Oct. 16)
 
**Completion Deadline:** Oct. 17

# **Documentation**
[Quests Description] (https://github.com/UQcsse3200/2024-studio-2/wiki/Quests)

# **Member**
Reemzheem Hossain (@reemmi)
Maree Robinson (@MargoCode)

### PR DESCRIPTION
**Pull request by MargoCode**
# Description

Ensuring that all quests work in all 3 biomes (mainly fixing minigames, final boss summoning after mini game, and final boss defeat listener). Redesigned pause menu (currently can't see due to buttons covering it) and edited quest tracker.  Plus small fixes to code style and javadocs. 

Pic of pause menu: (The text colour has been changed)
Screenshot 2024-10-12 at 1 30 30 pm

Fixes / Closes #635 #614 #516 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?
- [ ] Completing all of the quests in all 3 biomes (following hints and such from quest display (open by pressing 'q' in game) as they are unlocked)
- [ ] Checking redesign of pause menu (pressing 'esc' in game)

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by EllaBerglas** [comment]
Your pause menu doesn't work with the minigames like the current one in main

File: source/core/src/main/com/csse3200/game/overlays/PlayerStatsDisplay2.java [hunk 1]
```diff
@@ -69,14 +69,6 @@ private void addStatRow(Table table, String name, int value) {
...
-    private void addStatRow(Table table, String name, float value) {
-        Label nameLabel = new Label(name, skin, DEFAULT_TEXT);
-        Label valueLabel = new Label(String.valueOf(value), skin, DEFAULT_TEXT);
-        table.add(nameLabel).expandX().left();
-        table.add(valueLabel).expandX().right().padLeft(25f);
-        table.row().padTop(5f);
-    }
-
     private Table makeMenuBtns() {
         TextButton exitBtn = new TextButton("Leave Menu", skin);
```
Comment (yessica-s) [inline source/core/src/main/com/csse3200/game/overlays/PlayerStatsDisplay2.java:73]: potentially javadoc could be added to some of the functions in this class

**Review by yessica-s — APPROVED** [review]
Codewise looks good (I didn’t run it) good stuff :)

File: source/core/src/main/com/csse3200/game/overlays/PauseDisplay.java
Comment (EllaBerglas) [inline source/core/src/main/com/csse3200/game/overlays/PauseDisplay.java:1]: As I commented, does not work with minigames

**Review by EllaBerglas — CHANGES_REQUESTED** [review]

**Comment by EllaBerglas** [comment]
Is this what it is supposed to look like? 
image

**Review by nreyana — APPROVED** [review]
looks good :p haven't been able to access the mini-games in game so I've only checked the pause screen through the main menu, but everything works

**Pull request merged**